packages/dds/tree/src/core/tree/visitDelta.ts

```diff
@@ -249,8 +249,12 @@ function transferRoots(
  */
 export interface DeltaVisitor {
 	/**
-	 * Frees/releases the visitor. Must be called once the visitor is no longer needed, since trying to acquire
-	 * a new one before freeing an existing one is invalid.
+	 * Frees/releases the visitor.
+	 *
+	 * Must be called once the visitor finished traversing the delta for a couple of reasons:
+	 *
+	 * 1. Some visitors, such as those from forests, are put into a special mode while they have a visitor, forbidding some actions (like making more visitors).
+	 * 2. Some visitors, such as those from an anchorSet, defer some events for batching purposes until the visitor is freed.
```

Comment:
Nit (to ensure this line gets treated as a separate list entry)
```suggestion
	 *
	 * 2. Some visitors, such as those from an anchorSet, defer some events for batching purposes until the visitor is freed.
```